workflows/argo/resolve-output-url.yaml

```
@@ -11,20 +11,15 @@ spec:
           - name: bucket
           - name: project
           - name: tag
-      script:
-        image: debian:9.4
-        command: [bash]
-        source: |
-          set -e
-          fail_on_slash () {
-            if [[ "$1" == *"/"* ]]; then
-              echo "Slashes not permitted in bucket, project, and tag: $1"
-              exit 1
-            fi
-          }
-          DATE=$(date +'%F')
-          fail_on_slash {{inputs.parameters.bucket}}
-          fail_on_slash {{inputs.parameters.project}}
-          fail_on_slash {{inputs.parameters.tag}}
-          echo "gs://{{inputs.parameters.bucket}}/{{inputs.parameters.project}}/${DATE}/{{inputs.parameters.tag}}"
+      outputs:
+        parameters:
+        - name: url
+          valueFrom:
+            path: /tmp/url.txt
+      container:
+        image: us.gcr.io/vcm-ml/fv3net
+        command: ["bash", "-c", "-x", "-e"]
+        args:
+          - |
+            artifacts generate "{{inputs.parameters.bucket}}" "{{inputs.parameters.project}}" "{{inputs.parameters.tag}}" > /tmp/url.txt
           
```

I think you can access the stdout via `outputs.result` as the original code did and can simplify this a bit:
```
 ```suggestion
        command:
          - artifacts
          - generate
          - "{{inputs.parameters.bucket}}"
          - "{{inputs.parameters.project}}"
          - "{{inputs.parameters.tag}}"
          
```

See these argo docs: https://argoproj.github.io/argo-workflows/fields/#fields_23.

---

Oh I thought the `outputs.result` was only available for `script` templates. I'll give this a shot.